assets/js/phoenix_live_view/live_socket.js

```diff
@@ -366,9 +369,13 @@ export default class LiveSocket {
     }
   }
 
+  viewSelector(){
+    return `${PHX_VIEW_SELECTOR}${this.rootViewSelector || ""}`
```

Comment:
Should this be inversed, so people could do `#foo-bar`as the root view selector if they want to? In such cases, should they end it with a space?

Reply:
I was thinking it would simplify things if we just ignore the `PHX_VIEW_SELECTOR` and only use the user-defined selector?

The initial idea behind composing both together was because I didn't notice we could customise the `container` of the LiveView. So my initial implementation had something like: `:has(.some-class-inside-the-lv-template)`.

Reply:
Any reason not to just replace `PHX_VIEW_SELECTOR` completely with a user defined selector?

```javascript
new LiveSocket("/live", Socket, {params: {viewSelector: ".my-selector"}})
```

We can document users need to customise the LiveView `:container` in that case. It can be done globally in `use MyAppWeb, :liveview` or easily extend to accept as param `use MyAppWeb.LiveView, :liveview, app: :something` 🤔

Reply:
I think that would be my vote but we need to tell people they will want to check for [data-phx-session] unless they point explicitly to a selector.